tests/console/hwloc_testsuite.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's hwloc tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run hwloc testsuite
+# Maintainer: Thomas Blume <tblume@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+
+sub run() {
+    # install hwloc testsuite
+    script_run('zypper ar http://download.suse.de/ibs/QA:/SLE12SP2/update/ hwloc-testrepo');
```

Comment:
better use util function `zypper_call` for this and following zypper commands
see  here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/utils.pm#L295

and don't forget to `use utils;` in the beginning :)